### PR TITLE
[AAP-69139] Update data generator for proper perf-testing

### PR DESCRIPTION
**Pull request by ShaiahWren**
[AAP-69139](https://redhat.atlassian.net/browse/AAP-69139)

## Description
Adds missing data to the tools/anonymized_db_perf_data performance test data generator so collectors have realistic data to run against during perf testing.

**Part 1**: Adds a main_instance row so the controller_version_service collector returns actual controller version data instead of empty results. 

  Changes
  - helpers.py — new create_instance() function
  - fill_perf_db_data.py — calls create_instance() once during fill_init_data()     

**Part 2**: Adds 100 execution environment rows with 50 randomly
  generated installed collections each, and sets installed_collections on each job matching its EE. Jobs cycle through EEs so all collection sets are represented. This exercises jobs_by_installed_collections_versions in the rollup output.

Changes
 - helpers.py — new create_execution_environment(), create_execution_environments(), and _random_installed_collections() functions; removed fixed
  SAMPLE_INSTALLED_COLLECTIONS constant                        
  - fill_perf_db_data.py — creates 100 EEs during fill_init_data(), jobs cycle through them with matching installed_collections 

**Part 3**: Adds credentials linked to each job using the 4 built-in credential types already seeded in the DB (Machine, Vault, Amazon Web Services, Network). This exercises rollup_period_credential_types in the rollup output.

Changes:
- helpers.py — new create_credentials() and  create_job_credentials() functions          
-  fill_perf_db_data.py — creates credentials during fill_init_data(), links all 4 built-in types to each job

Extra changes:
Also adds --no-events flag to skip event generation.     
- Fixed duplicate key errors (main_instance hostname and main_executionenvironment name) when the generator is run multiple times without cleaning — names now include the run's unique suffix
                  

## Test plan

  - Clean existing data: python tools/anonymized_db_perf_data/clean_all_data.py
  - Fill with test data: python tools/anonymized_db_perf_data/fill_perf_db_data.py        
  --job-count=1000 --host-count=20 --no-events
  - Confirm instance row: SELECT version, node_type, enabled FROM main_instance; →  expect 4.5.0 | control | t
  - Confirm EEs: SELECT COUNT(*) FROM main_executionenvironment; → expect 100 rows  
  - Confirm jobs have installed_collections and execution_environment_id set: SELECT id, execution_environment_id, installed_collections FROM main_unifiedjob LIMIT 5;        
  - Confirm credentials linked to jobs: SELECT ct.name FROM main_unifiedjob_credentials ujc JOIN main_credential c ON c.id = ujc.credential_id JOIN main_credentialtype ct ON ct.id = c.credential_type_id LIMIT 4; → expect Machine, Vault, Amazon Web Services, Network
  - Run the rollup and confirm controller_versions, rollup_period_credential_types, and jobs_by_installed_collections_versions are all non-empty in the output

### Steps to Test

1.
2.
3.

## Required Actions

- [ ] Requires documentation updates

- [ ] Requires downstream repository changes

- [ ] Requires infrastructure/deployment changes

- [ ] Requires coordination with other teams

- [ ] Blocked by PR/MR: #XXX

### Code Quality

- [ ] Code is properly linted and formatted.
- [ ] All tests (existing and new) pass successfully.
- [ ] Tests are added or updated as needed.
- [ ] Code includes relevant comments for complex sections.
- [ ] Changes are reviewed and approved by at least two team members.
- [ ] Documentation is updated where applicable.  
       - If updates are required in the [handbook](https://github.com/ansible/handbook), create a separate PR for the handbook repo.

## Notes for Reviewers

Add any additional context or instructions for reviewers here - for example screenshots if needed.

### CODE REVIEW

File: tools/anonymized_db_perf_data/helpers.py [hunk 1]
```diff
@@ -723,5 +735,93 @@ def create_job_events(job_id, host_ids, task_count=50, job_index=0, job_created=
     print(f'Created {len(values)} job events ({task_count} tasks x {host_count} hosts)')
 
 
+# Sample installed_collections per EE. Jobs sharing the same EE must use the
+# same value — the rollup caches parsed collections once per execution_environment_id.
+SAMPLE_INSTALLED_COLLECTIONS = [
```
Comment (MilanPospisil) [inline tools/anonymized_db_perf_data/helpers.py]: This should be much larger for perf testing.

Try to use around 100 execution environments with random 50 installed collections.

Im not sure about correct numbers for customers, maybe noone knows. But we should test large edge case.

Comment (ShaiahWren) [reply tools/anonymized_db_perf_data/helpers.py]: Updated.

File: tools/anonymized_db_perf_data/helpers.py [hunk 2]
```diff
@@ -723,5 +735,99 @@ def create_job_events(job_id, host_ids, task_count=50, job_index=0, job_created=
...
+
+
+def create_execution_environment(name, image):
+    """Create an execution environment row and return its ID."""
+    sql = f"""
+    INSERT INTO main_executionenvironment (
+        created, modified, name, description, image, managed, pull
+    )
+    VALUES (
+        NOW(), NOW(), '{name}', '', '{image}', FALSE, 'missing'
```
Comment (hunterkepley) [inline tools/anonymized_db_perf_data/helpers.py:760]: _**technically**_ this is vulnerable to SQL injections  - i know this more than likely will never be used by any user, and the data more than likely will never be anything more than pre-made test data

but if we did want to be safe, this guide is a good one https://bobby-tables.com/python.html

**Review by hunterkepley — COMMENTED** [review]
looks good overall, left one comment about SQL injection vulnerabilities, but LGTM even if we decide against preventing those

File: tools/anonymized_db_perf_data/helpers.py (hunk 2 above)
Comment (ShaiahWren) [reply tools/anonymized_db_perf_data/helpers.py:760]: thanks for the review @hunterkepley  ! I'll look into addressing this as it is good practice.

Comment (ShaiahWren) [reply tools/anonymized_db_perf_data/helpers.py:760]: So, looking into it it looks like changing from f-strings to parameterized queries would involve all the create_ functions which would be quite a refactor. Since everything is internally generated I think our best bet _is_ to leave it.

Comment (hunterkepley) [reply tools/anonymized_db_perf_data/helpers.py:760]: sounds good to me

**Review by hunterkepley — APPROVED** [review]
LGTM (I dont have write access but i approved)
